File: session-05-chaos-engineering/java/src/main/java/coffeeshop/order/OrderService.java
package coffeeshop.order;

import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class OrderService {

    private final OrderRepository repository;
    private final coffeeshop.kitchen.KitchenService kitchenService;

    public OrderService(OrderRepository repository, coffeeshop.kitchen.KitchenService kitchenService) {
        this.repository = repository;
        this.kitchenService = kitchenService;
    }

    public List<Order> listOrders() {
        return repository.findAll();
    }

    public Optional<Order> findOrder(int id) {
        return repository.findById(id);
    }

    public coffeeshop.kitchen.Preparation placeOrder(int orderId, String drink) {
        return kitchenService.prepare(orderId, drink);
    }
}

File: session-06-capstone/typescript/src/order/order.service.ts
import {Injectable} from "@nestjs/common";
import {Order} from "./order.model";
import {OrderRepository} from "./order.repository";
import {KitchenService, Preparation} from "../kitchen/kitchen.service";

@Injectable()
export class OrderService {
    constructor(
        private readonly repository: OrderRepository,
        private readonly kitchenService: KitchenService,
    ) {
    }

    listOrders(): Order[] {
        return this.repository.findAll();
    }

    findOrder(id: number): Order | undefined {
        return this.repository.findById(id);
    }

    placeOrder(orderId: number, drink: string): Preparation {
        return this.kitchenService.prepare(orderId, drink);
    }
}
